presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTimeWithTimeZone.java

```diff
@@ -64,19 +60,17 @@
     public SqlTimeWithTimeZone(long timeWithTimeZone)
     {
         millisUtc = unpackMillisUtc(timeWithTimeZone);
-        if (millisUtc > MAX_MILLIS_TIME) {
-            throw new RuntimeException("millisUtc must be time in millisecond that passed since midnight UTC (" + millisUtc + " is to high)");
-        }
         timeZoneKey = unpackZoneKey(timeWithTimeZone);
+
+        validateMillisRange();
```

Comment:
if you did `this(unpackMillisUtc(timeWithTimeZone), unpackZoneKey(timeWithTimeZone))` you would be able to retain input validation directly in the (other) ctor.